lte/gateway/python/magma/mobilityd/uplink_gw.py

```diff
@@ -10,31 +10,42 @@
 See the License for the specific language governing permissions and
 limitations under the License.
 """
+import ipaddress
 import logging
 import netifaces
 
-from typing import MutableMapping, Optional
+from typing import MutableMapping, Optional, List
+from lte.protos.mobilityd_pb2 import GWInfo, IPAddress
+
+NO_VLAN = "NO_VLAN"
+
+
+def _get_vlan_key(vlan):
+    if vlan is None or vlan == '' or vlan == NO_VLAN:
+        return NO_VLAN
+    if int(vlan) < 0 or int(vlan) > 4096:
+        raise InvalidVlanId("invalid vlan: " + vlan)
+
+    return vlan
 
-DHCP_Router_key = "DHCPRouterKey"
-DHCP_Router_Mac_key = "DHCPRouterMacKey"
 
 # TODO: move helper class to separate directory.
 class UplinkGatewayInfo:
-    def __init__(self, gw_info_map: MutableMapping[str, str]):
+    def __init__(self, gw_info_map: MutableMapping[str, GWInfo]):
         """
             This maintains uptodate information about upstream GW.
 
         Args:
             gw_info_map: map to store GW info.
         """
         self._backing_map = gw_info_map
-        self._current_ip = None
-        self._current_mac = None
 
-    def getIP(self) -> Optional[str]:
-        if DHCP_Router_key in self._backing_map:
-            self._current_ip = self._backing_map.get(DHCP_Router_key)
-            return self._current_ip
+    def get_gw_ip(self, vlan_: str = "") -> Optional[str]:
```

Comment:
maybe rename to `vlan_str` ?

Reply:
actually vlan_str is confusing, so I changed it to vlan_id